src/DataAccess/ModerationReasonRepository.php

```diff
@@ -9,11 +9,14 @@
 
 class ModerationReasonRepository {
 
-	public function __construct(
-		private EntityManager $entityManager
-	) {
+	public function __construct( private EntityManager $entityManager ) {
 	}
 
+	/**
+	 * @param ModerationReason ...$moderationReasons
+	 *
+	 * @return array<string, ModerationReason>
```

Comment:
The reason why this fails is that the default result of the query builder query is `mixed`. We might get around that by installing the `phpstan-doctrine` extension and  replacing

```
$queryBuilder = $this->entityManager->createQueryBuilder();
$queryBuilder->select( 'mr' )
	->from( ModerationReason::class, 'mr' );
```
with

```
$queryBuilder = $this->getEntityManager()->getRepository(ModerationReason::class)
            ->createQueryBuilder('mr');
```

Reply:
@gbirke This was not needed. Following solved the error:

```
		/** @var ModerationReason[] $result */
		$result = $query->getResult();

		return $result;
```